# Livro para consulta:
- https://jakevdp.github.io/PythonDataScienceHandbook/03.08-aggregation-and-grouping.html
- https://jakevdp.github.io/PythonDataScienceHandbook/03.09-pivot-tables.html
    

# 1. Importando bibliotecas <a name="import"></a>

<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline

# 2. Carregando o dataframe SINASC <a name="read"></a>
<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [2]:
sinasc_raw = pd.read_csv('SINASC_RO_2019.csv')
sinasc_raw.head()

ORIGEM   CODESTAB  CODMUNNASC  LOCNASC  IDADEMAE  ESTCIVMAE  \
0       1  2679477.0      110001        1        19        5.0   
1       1  2679477.0      110001        1        29        2.0   
2       1  2679477.0      110001        1        37        9.0   
3       1  2516500.0      110001        1        30        5.0   
4       1  2516500.0      110001        1        30        2.0   

            ESCMAE  CODOCUPMAE  QTDFILVIVO  QTDFILMORT  ...  KOTELCHUCK  \
0      8 a 11 anos         NaN         0.0         0.0  ...           5   
1      8 a 11 anos    999992.0         1.0         0.0  ...           5   
2      8 a 11 anos    513205.0         2.0         0.0  ...           5   
3  12 anos ou mais    231205.0         0.0         0.0  ...           4   
4      8 a 11 anos    999992.0         1.0         0.0  ...           5   

  CONTADOR munResStatus munResTipo               munResNome  munResUf  \
0        1        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
1        2        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
2        3        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
3        4        ATIVO      MUNIC  Alto Alegre dos Parecis  Rondônia   
4        5        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   

   munResLat munResLon  munResAlt  munResArea  
0  -11.93554 -61.99982      338.0    7067.025  
1  -11.93554 -61.99982      338.0    7067.025  
2  -11.93554 -61.99982      338.0    7067.025  
3  -12.13178 -61.85308      397.0    3958.273  
4  -11.93554 -61.99982      338.0    7067.025  

[5 rows x 69 columns]

In [3]:
sinasc_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27028 entries, 0 to 27027
Data columns (total 69 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   ORIGEM        27028 non-null  int64  
 1   CODESTAB      26913 non-null  float64
 2   CODMUNNASC    27028 non-null  int64  
 3   LOCNASC       27028 non-null  int64  
 4   IDADEMAE      27028 non-null  int64  
 5   ESTCIVMAE     26711 non-null  float64
 6   ESCMAE        26716 non-null  object 
 7   CODOCUPMAE    24121 non-null  float64
 8   QTDFILVIVO    25455 non-null  float64
 9   QTDFILMORT    24930 non-null  float64
 10  CODMUNRES     27028 non-null  int64  
 11  GESTACAO      25796 non-null  object 
 12  GRAVIDEZ      26949 non-null  object 
 13  PARTO         26979 non-null  object 
 14  CONSULTAS     27028 non-null  int64  
 15  DTNASC        27028 non-null  object 
 16  HORANASC      27007 non-null  float64
 17  SEXO          27024 non-null  object 
 18  APGAR1        26932 non-nu

In [4]:
sinasc_raw.drop_duplicates(inplace=True)
sinasc_raw.shape

(27028, 69)

# Tarefa 1

### 1. Idade media das mães e dos pais por município (coluna munResNome)


In [5]:
sinasc_1 = sinasc_raw[['munResNome','IDADEMAE','IDADEPAI']]
sinasc_1.head()

munResNome  IDADEMAE  IDADEPAI
0    Alta Floresta D'Oeste        19      26.0
1    Alta Floresta D'Oeste        29      24.0
2    Alta Floresta D'Oeste        37      32.0
3  Alto Alegre dos Parecis        30      24.0
4    Alta Floresta D'Oeste        30      27.0

In [6]:
sinasc_1.groupby('munResNome')
sinasc_1.groupby('munResNome').mean()

IDADEMAE   IDADEPAI
munResNome                                     
Alta Floresta D'Oeste      25.991826  29.548701
Alto Alegre dos Parecis    24.844156  29.179104
Alto Paraíso               24.960177  28.833333
Alvorada D'Oeste           25.770732  30.757282
Ariquemes                  25.607866  32.466667
Buritis                    25.559177  30.812500
Cabixi                     26.037500  34.333333
Cacaulândia                25.546667  36.333333
Cacoal                     26.890102  30.931854
Campo Novo de Rondônia     24.825301  30.500000
Candeias do Jamari         25.243323  29.894737
Castanheiras               27.281250  30.392857
Cerejeiras                 27.240506  31.652778
Chupinguaia                25.333333  29.629630
Colorado do Oeste          27.602804  34.354839
Corumbiara                 24.716535  32.470588
Costa Marques              24.412017  30.613636
Cujubim                    24.482927  31.400000
Espigão D'Oeste            26.093333  31.198113
Governador Jorge Teixeira  24.401786  29.222222
Guajará-Mirim              25.462228  30.470588
Itapuã do Oeste            26.064220  42.500000
Jaru                       25.982974  32.197183
Ji-Paraná                  26.361137  31.132239
Machadinho D'Oeste         25.480000  31.326923
Ministro Andreazza         27.904762  33.849315
Mirante da Serra           26.304878  32.154762
Monte Negro                25.648241  34.678571
Município ignorado - RO    24.000000  22.000000
Nova Brasilândia D'Oeste   26.222222  30.520833
Nova Mamoré                25.326389  31.692308
Nova União                 26.894231  31.371429
Novo Horizonte do Oeste    26.938776  31.588235
Ouro Preto do Oeste        26.375594  31.233244
Parecis                    26.454545  32.473684
Pimenta Bueno              25.199675  29.745174
Pimenteiras do Oeste       25.075000  35.600000
Porto Velho                26.312670  32.352679
Presidente Médici          25.796053  30.150754
Primavera de Rondônia      27.023256  31.888889
Rio Crespo                 26.660000  28.000000
Rolim de Moura             26.600863  30.360479
Santa Luzia D'Oeste        26.121212  30.703297
Seringueiras               24.910714  30.500000
São Felipe D'Oeste         26.481481  31.395833
São Francisco do Guaporé   25.660066  31.638889
São Miguel do Guaporé      25.926346  32.666667
Teixeirópolis              26.031250  31.500000
Theobroma                  25.536082  30.083333
Urupá                      26.688235  31.008929
Vale do Anari              26.096774  34.000000
Vale do Paraíso            25.679012  29.976744
Vilhena                    26.433962  34.152941

### 2. Peso médio dos bebes por sexo que nasceram no dia do seu aniversário por faixas de escolaridade mae
Ex: Você, aluna(o), nasceu no dia 10/01, então você precisa filtrar o conjunto de dados nessa data e calcular o peso médio dos bebês de cada sexo por faixa de escolaridade da mãe.

In [7]:
sinasc_raw['DTNASC'].info

<bound method Series.info of 0        2019-02-19
1        2019-02-21
2        2019-02-25
3        2019-03-20
4        2019-03-23
            ...    
27023    2019-12-13
27024    2019-10-04
27025    2019-08-02
27026    2019-12-23
27027    2019-12-27
Name: DTNASC, Length: 27028, dtype: object>

In [8]:
aniversario = sinasc_raw.loc[sinasc_raw['DTNASC'] == '2019-05-27']
aniversario.shape

(100, 69)

In [9]:
sinasc_2 = aniversario[['ESCMAE','PESO','SEXO']]
sinasc_2

ESCMAE  PESO       SEXO
946    12 anos ou mais  3290  Masculino
947    12 anos ou mais  3695  Masculino
948    12 anos ou mais  3180   Feminino
983        8 a 11 anos  3830  Masculino
984        8 a 11 anos  3565  Masculino
...                ...   ...        ...
13128      8 a 11 anos  3230  Masculino
13135  12 anos ou mais  3265   Feminino
13537      8 a 11 anos  3725  Masculino
13701      8 a 11 anos  2730   Feminino
13702      8 a 11 anos  3180   Feminino

[100 rows x 3 columns]

In [10]:
sinasc_2.groupby(['ESCMAE','SEXO']).mean()

PESO
ESCMAE          SEXO                  
1 a 3 anos      Masculino  3475.000000
12 anos ou mais Feminino   3201.666667
                Masculino  3332.214286
4 a 7 anos      Feminino   3399.166667
                Masculino  3491.250000
8 a 11 anos     Feminino   3249.068966
                Masculino  3279.464286

### 3. Qual o municipio que nasceu menos bebe em 2019?
    - qual a idade media, maxima, minima das maes nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?

In [11]:
sinasc_raw['munResNome'].value_counts()

Porto Velho                  8437
Ji-Paraná                    2182
Ariquemes                    1729
Vilhena                      1590
Cacoal                       1374
Rolim de Moura                927
Jaru                          881
Guajará-Mirim                 781
Ouro Preto do Oeste           631
Pimenta Bueno                 616
Buritis                       583
Machadinho D'Oeste            500
Espigão D'Oeste               450
Nova Mamoré                   432
Alta Floresta D'Oeste         367
São Miguel do Guaporé         353
Candeias do Jamari            337
Presidente Médici             304
São Francisco do Guaporé      303
Nova Brasilândia D'Oeste      261
Cerejeiras                    237
Costa Marques                 233
Alto Paraíso                  226
Colorado do Oeste             214
Alvorada D'Oeste              205
Cujubim                       205
Monte Negro                   199
Urupá                         170
Seringueiras                  168
Campo Novo de 

O município de Castanheiras teve o menor número de nascimentos de 2019 (32)

In [12]:
sinasc_3 = sinasc_raw.loc[(sinasc_raw['munResNome'] == 'Castanheiras'), ['munResNome','IDADEMAE','IDADEPAI']]
sinasc_3.dropna()

munResNome  IDADEMAE  IDADEPAI
2067   Castanheiras        18      17.0
2300   Castanheiras        31      36.0
5881   Castanheiras        26      24.0
5952   Castanheiras        32      29.0
10762  Castanheiras        21      19.0
10768  Castanheiras        31      31.0
10882  Castanheiras        23      25.0
10962  Castanheiras        29      32.0
10988  Castanheiras        31      42.0
10997  Castanheiras        35      38.0
11017  Castanheiras        27      32.0
11175  Castanheiras        24      28.0
11196  Castanheiras        27      35.0
11293  Castanheiras        25      29.0
11354  Castanheiras        34      38.0
17441  Castanheiras        39      43.0
24282  Castanheiras        28      29.0
24288  Castanheiras        27      35.0
24299  Castanheiras        30      35.0
24486  Castanheiras        35      38.0
24507  Castanheiras        27      27.0
24517  Castanheiras        27      20.0
24528  Castanheiras        25      32.0
24605  Castanheiras        32      25.0
24639  Castanheiras        18      27.0
24715  Castanheiras        32      33.0
24781  Castanheiras        22      19.0
24863  Castanheiras        30      33.0

In [13]:
sinasc_3.groupby(['munResNome']).agg([('média','mean'),'max','min'])

IDADEMAE           IDADEPAI            
                 média max min      média   max   min
munResNome                                           
Castanheiras  27.28125  39  17  30.392857  43.0  17.0

### 4. Qual o municipio que nasceu mais bebe no mês de março?
    - qual a quantidade de filhos vivos media, maxima, minima nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?



In [14]:
mes_nasc = []

data_nasc = list(sinasc_raw.DTNASC)

for i in data_nasc:
    x = i.split('-')
    mes_nasc.append(x[1])
    
sinasc_raw['mes_nasc'] = mes_nasc

sinasc_raw.mes_nasc

0        02
1        02
2        02
3        03
4        03
         ..
27023    12
27024    10
27025    08
27026    12
27027    12
Name: mes_nasc, Length: 27028, dtype: object

In [15]:
sinasc_4 = sinasc_raw.loc[(sinasc_raw['mes_nasc'] == '03'), ['munResNome','QTDFILVIVO','IDADEPAI']]
sinasc_4

munResNome  QTDFILVIVO  IDADEPAI
3      Alto Alegre dos Parecis         0.0      24.0
4        Alta Floresta D'Oeste         1.0      27.0
5        Alta Floresta D'Oeste         2.0      40.0
9        Alta Floresta D'Oeste         1.0      41.0
10       Alta Floresta D'Oeste         1.0      28.0
...                        ...         ...       ...
13670              Porto Velho         0.0       NaN
13671              Porto Velho         2.0       NaN
13736              Porto Velho         1.0      42.0
13748                  Vilhena         2.0       NaN
13750                Ji-Paraná         0.0      50.0

[2456 rows x 3 columns]

In [16]:
sinasc_4['munResNome'].value_counts()

Porto Velho                  744
Ji-Paraná                    188
Vilhena                      148
Ariquemes                    141
Cacoal                       133
Jaru                          80
Guajará-Mirim                 75
Rolim de Moura                69
Ouro Preto do Oeste           67
Pimenta Bueno                 62
Buritis                       59
Nova Mamoré                   53
Machadinho D'Oeste            44
Espigão D'Oeste               44
São Miguel do Guaporé         41
Candeias do Jamari            33
São Francisco do Guaporé      31
Alta Floresta D'Oeste         31
Presidente Médici             29
Costa Marques                 24
Alto Paraíso                  22
Alvorada D'Oeste              22
Monte Negro                   20
Nova Brasilândia D'Oeste      20
Seringueiras                  19
Colorado do Oeste             19
Chupinguaia                   18
Cujubim                       17
Cerejeiras                    16
Urupá                         14
Vale do Pa

O município com o maior número de nascimentos no mês de março foi Porto Velho (744).

In [21]:
sinasc_4.dropna().loc[(sinasc_4['munResNome'] == 'Porto Velho')].groupby(['munResNome']).agg([('média','mean'),'max','min'])

QTDFILVIVO           IDADEPAI            
                 média  max  min    média   max   min
munResNome                                           
Porto Velho   1.479167  6.0  0.0    35.75  62.0  19.0

### Analise as respostas encontradas, tire algum insight delas, conte pra gente algo encontrado nos dados. Algo que você julgue relevante e novo pra você.

- A idade média das mães de bebês nascidos no ano 2019 no município de Castanhares é menor que a dos pais.
- Teve um total de 2456 nascimentos no mês de março.
- Porto Velho teve o maior número de nascimentos em 2019.
- A idade máxima dos pais de bebês nascidos no mês de março, no município de Porto Velho, é quase o dobro da idade média.